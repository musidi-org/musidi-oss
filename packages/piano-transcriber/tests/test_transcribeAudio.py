from scripts import transcribeAudio


def transcribeFileSnapshot(audioFileName):
    audioFilePath = f"samples/{audioFileName}"
    midiFilePath = f"{audioFilePath}.mid"
    transcribeAudio.transcribeFile(audioFilePath)
    f = open(midiFilePath, "rb")
    return f.read()


def test_transcribe(snapshot):
    audioFiles = [
        "example.mp3",
        "example.webm",
        "example.ogg",
        "example.flac",
        "example.m4a",
        "example.opus",
    ]
    for audioFile in audioFiles:
        assert transcribeFileSnapshot(audioFile) == snapshot
