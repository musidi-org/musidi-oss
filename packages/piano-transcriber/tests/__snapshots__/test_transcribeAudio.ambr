# serializer version: 1
# name: test_transcribe_flac
  b'MThd\x00\x00\x00\x06\x00\x01\x00\x02\x01\x80MTrk\x00\x00\x00\x13\x00\xffQ\x03\x07\xa1 \x00\xffX\x04\x04\x02\x18\x08\x01\xff/\x00MTrk\x00\x00\x00V0\x909K\x00<K\x015J\x012C!9\x00\x025\x00\x01<\x00\x082\x00\x81.5D\x00+?\x027D\x02;H#+\x00\x037\x00\x005\x00\x1e;\x00\x81\x1f4B\x01<E\x000C\x007D#7\x00\x000\x00\x024\x00\x03<\x00\x81<$:($\x00\x01\xff/\x00'
# ---
# name: test_transcribe_m4a
  b'MThd\x00\x00\x00\x06\x00\x01\x00\x02\x01\x80MTrk\x00\x00\x00\x13\x00\xffQ\x03\x07\xa1 \x00\xffX\x04\x04\x02\x18\x08\x01\xff/\x00MTrk\x00\x00\x00V0\x909L\x015I\x00<J\x012B!9\x00\x035\x00\x00<\x00\x072\x00\x81/+?\x015B\x037C\x02;H +\x00"5\x00\x007\x00\x00;\x00\x81\x1f4A\x000B\x01<D\x007D"0\x00\x017\x00\x034\x00\x02<\x00\x81<$:\'$\x00\x01\xff/\x00'
# ---
# name: test_transcribe_mp3
  b'MThd\x00\x00\x00\x06\x00\x01\x00\x02\x01\x80MTrk\x00\x00\x00\x13\x00\xffQ\x03\x07\xa1 \x00\xffX\x04\x04\x02\x18\x08\x01\xff/\x00MTrk\x00\x00\x00V0\x909K\x005I\x00<K\x022B!9\x00\x025\x00\x00<\x00\x062\x00\x811+A\x005C\x027C\x01;G$+\x00\x027\x00\x015\x00\x1e;\x00\x81\x1f4A\x010C\x00<E\x007D$7\x00\x000\x00\x024\x00\x02<\x00\x81<$9)$\x00\x01\xff/\x00'
# ---
# name: test_transcribe_ogg
  b'MThd\x00\x00\x00\x06\x00\x01\x00\x02\x01\x80MTrk\x00\x00\x00\x13\x00\xffQ\x03\x07\xa1 \x00\xffX\x04\x04\x02\x18\x08\x01\xff/\x00MTrk\x00\x00\x00V0\x909J\x01<J\x005F\x012B!9\x00\x015\x00\x01<\x00\x082\x00\x81.+A\x015D\x027D\x01;H&+\x00\x035\x00\x01;\x00\x1b7\x00\x81\x1f4C\x01<F\x000C\x007E$0\x00\x027\x00\x024\x00\x01<\x00\x81<$:)$\x00\x01\xff/\x00'
# ---
# name: test_transcribe_opus
  b'MThd\x00\x00\x00\x06\x00\x01\x00\x02\x01\x80MTrk\x00\x00\x00\x13\x00\xffQ\x03\x07\xa1 \x00\xffX\x04\x04\x02\x18\x08\x01\xff/\x00MTrk\x00\x00\x00V0\x909K\x005J\x01<J\x002C$9\x00\x025\x00\x01<\x00\x042\x00\x81/+@\x015D\x027D\x01;H!+\x00\x065\x00\x017\x00\x1d;\x00\x81\x1f4B\x010B\x00<E\x007D&7\x00\x000\x00\x014\x00\x01<\x00\x81<$:*$\x00\x01\xff/\x00'
# ---
# name: test_transcribe_webm
  b'MThd\x00\x00\x00\x06\x00\x01\x00\x02\x01\x80MTrk\x00\x00\x00\x13\x00\xffQ\x03\x07\xa1 \x00\xffX\x04\x04\x02\x18\x08\x01\xff/\x00MTrk\x00\x00\x00V0\x909K\x005J\x01<J\x002C$9\x00\x025\x00\x01<\x00\x042\x00\x81/+@\x015D\x027D\x01;H!+\x00\x065\x00\x017\x00\x1d;\x00\x81\x1f4B\x010B\x00<E\x007D&7\x00\x000\x00\x014\x00\x01<\x00\x81<$:*$\x00\x01\xff/\x00'
# ---
